### PR TITLE
Add act.enterTtext()

### PR DESCRIPTION
**Pull request by rehlma**
Add the `tester.enterText()` functionality to Spot

```dart
testWidgets('enter text in text form field', (tester) async {
    await tester.pumpWidget(
      MaterialApp(home: Material(child: Form(child: TextFormField()))),
    );
    await act.enterText(spot<TextFormField>(), 'hello');
    spotText('hello').existsOnce();
});
```

### CODE REVIEW

File: test/act/act_test.dart [hunk 1]
```diff
@@ -13,231 +13,252 @@ void main() {
...
-  });
-
-  testWidgets('tap must be awaited', (tester) async {
-    await tester.pumpWidget(
-      MaterialApp(
-        home: Center(
-          child: ElevatedButton(
-            onPressed: () {},
-            child: const Text('Home'),
+  group('tap', () {
```
Comment (rehlma) [inline test/act/act_test.dart:16]: I just wrapped all `tap` tests in a group and added a new group for `enterText`. I didn't actually change any `tap` test here.

File: lib/src/act/act.dart [hunk 2]
```diff
@@ -18,6 +18,45 @@ const act = Act._();
...
+    final skipOffstage = selector.finder.skipOffstage;
+
+    return TestAsyncUtils.guard<void>(() async {
+      final binding = WidgetsBinding.instance as TestWidgetsFlutterBinding;
+
+      final editableText = find.descendant(
+        of: selector.finder,
+        matching: find.byType(EditableText, skipOffstage: skipOffstage),
+        matchRoot: true,
+      );
```
Comment (passsy) [inline lib/src/act/act.dart:35]: we can use spot here, right? 😉

File: lib/src/act/act.dart [hunk 3]
```diff
@@ -18,6 +18,45 @@ const act = Act._();
...
+
+      // Setting focusedEditable causes the binding to call requestKeyboard()
+      // on the EditableTextState, which itself eventually calls TextInput.attach
+      // to establish the connection.
+      binding.focusedEditable = editableTextState;
+      await binding.pump();
+
+      final testTextInput = binding.testTextInput;
+      testTextInput.enterText(text);
+      await binding.pump();
```
Comment (passsy) [inline lib/src/act/act.dart:56]: Compare with https://github.com/leancodepl/patrol/blob/9577d2e14598848226e9997283fd981bde727d3c/packages/patrol_finders/lib/src/custom_finders/patrol_tester.dart#L335-L354

File: lib/src/act/act.dart (hunk 2 above)
Comment (rehlma) [reply lib/src/act/act.dart:35]: I thought we want to use the Flutter implementation here but I'm sure we can use spot.

Comment (passsy) [reply lib/src/act/act.dart:35]: Flutter yes, but not the finder API. That's we we're replacing :)

**Comment by passsy** [comment]
I tried it, here's an example test for a login form.

```dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:f93_forming_forms/main.dart';
import 'package:spot/spot.dart';

final textFieldLabelText = widgetProp<TextField, String?>(
  'labelText',
  (textField) => textField.decoration?.labelText,
);

void main() {
  group('Part 1', () {
    testWidgets('enter username password', (tester) async {
      await tester.pumpWidget(FormingFormsKata());

      final email = spot<TextField>()
          .whereWidgetProp(textFieldLabelText, (label) => label == 'Email')
        ..existsOnce();
      await act.enterText(email, 'pascal@phntm.xyz');

      final password = spot<TextField>()
          .whereWidgetProp(textFieldLabelText, (label) => label == 'Password')
        ..existsOnce();
      await act.enterText(password, 'asdfasdf');

      await act.tap(spotText('Accept AGBs'));
      await tester.pump();

      // wait for async matchers and debounce
      await tester.pump(Duration(milliseconds: 250));
      await tester.pump(Duration(seconds: 2));

      await act.tap(spotText('Sign in'));
      await tester.pump();
      await tester.pump(Duration(seconds: 1));

      spotText('Success').existsOnce();
    });
  });
}
```

**Pull request merged**